src/context/filter.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{
+    Context, RedisModuleCommandFilter, RedisModuleCommandFilterCtx, RedisModuleString,
+    RedisModule_CommandFilterArgDelete, RedisModule_CommandFilterArgGet,
+    RedisModule_CommandFilterArgInsert, RedisModule_CommandFilterArgReplace,
+    RedisModule_CommandFilterArgsCount, RedisModule_CommandFilterGetClientId,
+    RedisModule_CreateString, RedisModule_RegisterCommandFilter,
+    RedisModule_UnregisterCommandFilter, ValkeyString,
+};
+use std::ffi::{c_int, CString};
+use std::ptr::null_mut;
+
+#[derive(Debug)]
+pub struct CommandFilter {
+    pub inner: *mut RedisModuleCommandFilter,
+}
+
+// otherwise you get error:     cannot be shared between threads safely
+unsafe impl Send for CommandFilter {}
+unsafe impl Sync for CommandFilter {}
+
+impl Context {
+    pub fn register_command_filter(
+        &self,
+        module_cmd_filter_func: extern "C" fn(*mut RedisModuleCommandFilterCtx),
+        flags: c_int,
+    ) -> CommandFilter {
+        let module_cmd_filter = unsafe {
+            RedisModule_RegisterCommandFilter.unwrap()(
+                self.ctx,
+                Some(module_cmd_filter_func),
+                flags,
+            )
+        };
+        CommandFilter {
+            inner: module_cmd_filter,
+        }
+    }
+    pub fn unregister_command_filter(&self, cmd_filter: &CommandFilter) {
+        unsafe {
+            RedisModule_UnregisterCommandFilter.unwrap()(self.ctx, cmd_filter.inner);
+        }
+    }
+}
+
+pub fn command_filter_args_count(ctx: *mut RedisModuleCommandFilterCtx) -> c_int {
+    unsafe { RedisModule_CommandFilterArgsCount.unwrap()(ctx) }
+}
+
+pub fn command_filter_arg_get(
+    ctx: *mut RedisModuleCommandFilterCtx,
+    pos: c_int,
+) -> *mut RedisModuleString {
+    unsafe { RedisModule_CommandFilterArgGet.unwrap()(ctx, pos) }
+}
+
+/// wrapper to get argument as a String instead of RedisModuleString
+pub fn command_filter_arg_get_as_string(
+    ctx: *mut RedisModuleCommandFilterCtx,
+    pos: c_int,
+) -> String {
+    let arg = command_filter_arg_get(ctx, pos);
+    ValkeyString::from_ptr(arg).unwrap().to_string()
```

Comment:
Instead of unwrapping here (panicking on utf errors), would it be better to return the Result back to the Module developer and let them handle this?
`ValkeyString::from_ptr` returns a `Result<&'a str, Utf8Error>` which we can return back

Reply:
agree, changed it.

Reply:
agree, changed it